extensions/wikia/DefaultMessages/DefaultMessagesCache.php

```diff
@@ -0,0 +1,521 @@
+<?php
+
+/**
+ * Default message cache (based on MW Message Cache)
+ * Stores messages from messaging.wikia.com
+ */
+class DefaultMessagesCache {
+	// Holds loaded messages that are defined in MediaWiki namespace.
+	var $mCache;
+
+	var $mUseCache, $mDisable, $mExpiry;
+
+	// Variable for tracking which variables are loaded
+	var $mLoadedLanguages = array();
+
+	var $mLocalMessageCache = false;
+	var $mLocalMessageCacheSerialized = true;
+
+	function __construct( &$memCached, $useDB, $expiry ) {
+		global $wgLocalMessageCache;
+
+		$this->mUseCache = !is_null( $memCached );
+		$this->mMemc = &$memCached;
+		$this->mDisable = !$useDB;
+		$this->mExpiry = $expiry;
+
+		if( $wgLocalMessageCache !== false ) {
+			$this->mLocalMessageCache = $wgLocalMessageCache;
+		}
+	}
+
+	private function memcKey( /*... */ ) {
+		global $wgDefaultMessagesDB;
+		$args = func_get_args();
+		$key = $wgDefaultMessagesDB . ':' . implode( ':', $args );
+		return $key;
+	}
+
+	/**
+	 * Try to load the cache from a local file.
+	 * Actual format of the file depends on the $wgLocalMessageCacheSerialized
+	 * setting.
+	 *
+	 * @param $hash String: the hash of contents, to check validity.
+	 * @param $code Mixed: Optional language code, see documenation of load().
+	 * @return false on failure.
+	 */
+	function loadFromLocal( $hash, $code ) {
+		global $wgDefaultMessagesDB;
+
+		$filename = $this->mLocalMessageCache . '/' . substr( $wgDefaultMessagesDB, 0, 1 ) . '/' . $wgDefaultMessagesDB . "/messages-$code";
+
+		# Check file existence
+		wfSuppressWarnings();
+		$file = fopen( $filename, 'r' );
+		wfRestoreWarnings();
+		if ( !$file ) {
+			return false; // No cache file
+		}
+
+		if ( $this->mLocalMessageCacheSerialized ) {
+			// Check to see if the file has the hash specified
+			$localHash = fread( $file, 32 );
+			if ( $hash === $localHash ) {
+				// All good, get the rest of it
+				$serialized = '';
+				while ( !feof( $file ) ) {
+					$serialized .= fread( $file, 100000 );
+				}
+				fclose( $file );
+				return $this->setCache( unserialize( $serialized ), $code );
+			} else {
+				fclose( $file );
+				return false; // Wrong hash
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)